A77-xds-rate-limiting-rlqs.md

```diff
@@ -0,0 +1,490 @@
+5. RLQS filter state will persist across LDS/RDS updates using cache retention
+   mechanism similar to the one implemented for [gRFC A83].
+
+### Related Proposals:
+
+* [A27: xDS-Based Global Load Balancing][gRFC A27]
+* [A36: xDS-Enabled Servers][gRFC A36]
+* [A39: xDS HTTP Filter Support][gRFC A39]
+* [A41: xDS RBAC Support][gRFC A41]
+* [A83: xDS GCP Authentication Filter][gRFC A83]
+
+[gRFC A27]: A27-xds-global-load-balancing.md
+[gRFC A41]: A41-xds-rbac.md
+[gRFC A36]: A36-xds-for-servers.md
+[gRFC A39]: A39-xds-http-filters.md
+[gRFC A83]: A83-xds-gcp-authn-filter.md
+
+[`GrpcService`]: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/grpc_service.proto
+[`GrpcService.GoogleGrpc`]: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/grpc_service.proto#envoy-v3-api-msg-config-core-v3-grpcservice-googlegrpc
+[Unified Matcher API]: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/matching/matching_api.html
+[Envoy CEL environment]: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes
+
+[rlqs_proto]: https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/rate_limit_quota/v3/rlqs.proto.html
+[rlqs_filter_proto]: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/rate_limit_quota/v3/rate_limit_quota.proto
+
+## Proposal
+
+### RLQS Components Overview
+
+The diagram below shows the conceptual components of the RLQS Filter. Note that
+the actual implementation may vary depending on the language.
+
+```mermaid
+graph TD
+%% RLQS Components Flowchart v7
+
+%% == nodes ==
+    rlqs_filter(RLQS HTTP Filter)
+    rlqs_cache(RLQS Cache)
+    rlqs_filter_state(RLQS Filter State)
+    rlqs_client(RLQS Client)
+    rlqs_server[(RLQS Server)]
+    rlqs_bucket_cache(RLQS Bucket Cache)
+    report_timers(Report Timers)
+    matcher_tree(Matcher Tree)
+    rlqs_bucket(RLQS Bucket)
+    rpc_handler("Filter's onClientCall handler")
+    request{{RPC}}
+%% == edges ==
+    rlqs_filter -- " Get RLQS Filter State<br />per unique config " --> rlqs_cache -- " getOrCreate(config) " --> rlqs_filter_state
+    rlqs_filter -- " Pass RLQS Filter State<br />for the route " --> rpc_handler -- " rateLimit(call) " --> rlqs_filter_state
+    request --> rpc_handler
+    rlqs_filter_state --o matcher_tree & report_timers
+    rlqs_filter_state -- sendUsageReports --> rlqs_client
+    rlqs_filter_state -- CRUD --> rlqs_bucket_cache
+    rlqs_client -- onBucketsUpdate --> rlqs_filter_state
+    rlqs_client <-. gRPC Stream .-> rlqs_server
+    rlqs_bucket_cache -- " getOrCreate(bucketId)<br />Atomic Updates " --> rlqs_bucket
+    style request stroke: RoyalBlue, stroke-width: 2px;
+    linkStyle 3,4 stroke: RoyalBlue, stroke-width: 2px;
+```
+
+##### RLQS HTTP Filter
+
+The filter parses the config, combines LDS filter config with RDS overrides, and
+generates the `onClientCall` handlers (aka interceptors in Java and Go, and
+filters in C++).
+
+##### RLQS Cache
```

Comment:
Yes, I meant A83-based cache here, I've pointed it out in several places. This was supposed to describe how this persistent cache is _used_ by RLQS. I see how this can be confusing for the reader. Will update accordingly.